### ISSUE TITLE
Implement filter refinement for AOs

### ISSUE THREAD

**Issue by porta-antiporta** [18F/fec-eregs#276]
Based off #274 

The initial findings for implementation of search and refinement for AOs is the following fields is needed.
- [ ] AO number (search box)
- [ ] AO Issued Date (date range)
- [ ] AO Name (search box)
- [ ] Entity Type (conditional drop-down menu)  NOTE: this is only really useful if it can be combined with other search criteria (currently cannot be combined with text search terms)
- [ ] Entity Name (search box)
- [ ] Browse by year (links to results, confined to selected year)
- [ ] Pending AO Requests (links to a list of the pending requests)

Discuss in grooming which of these makes sense to be done together

**Comment by porta-antiporta** [comment]
From a dev perspective:

- Milestone for all the data we already have
- Entity Name, AO Date, AO Number

- Milestone for new data that the API does not yet have.

**Comment by jenniferthibault** [comment]
@anthonygarvan this is ready

**Comment by adrianwebb** [comment]
Split issue into multiple with @anthonygarvan

**Comment by adrianwebb** [comment]
Design needs feedback before we start development

**Comment by jenniferthibault** [comment]
@adrianwebb / @anthonygarvan / @nickykrause  Sorry for the confusion, but what feedback do we (design) need? I'm sure i missed this from discussion during sprint planning, so apologies for the redundancy, but it would be helpful to be specific and capture it in this issue—I thought it was ready to go.

**Comment by adrianwebb** [comment]
I believe this feedback was directed at getting feedback from Amy / Wei before we started development, just to make sure they understood the design.  Let me know if they already reviewed this and discussed their thoughts with you.

**Comment by jenniferthibault** [comment]
Thanks for the clarification @adrianwebb !

 @amykort , can you take a look at the first round of filters for Advisory Opinions, mocked up in [the above comment](https://github.com/18F/fec-eregs/issues/276#issuecomment-255891751)?

One thing @emileighoutlaw and I had been wondering particularly is if we could use the word `Participants` instead of `Entity` for the category that includes the Requestor, the Counsel, and the Commenter.

Anything else that pops out at you, welcome and open for discussion!

If you want to see how the tags will be generated, check out [this mockup too](https://github.com/18F/openFEC-web-app/issues/1619#issuecomment-254844282).

**Comment by AmyKort** [comment]
Are we trying to make the filter categories the same across all types of resources (AOs, MURs, and ADR cases, for example) or will the filter categories change with each type of resource?

**Comment by jenniferthibault** [comment]
The categories will most definitely change with each type of resource—although we've found that there are common factors to how people narrow down legal documents, like date and document type. 

So when we introduce filters to MURs you may see some differences in the filters we're able to provide, but we'd like to keep as many common components as it makes sense to, which would make a better, easier searching experience.

**Comment by AmyKort** [comment]
I have a couple of questions, but we really want Policy’s input.

What’s the reasoning behind surfacing commissioner statements outside the “more” box, and making commissioner statements the second choice—rather than requests or something else? I counted only 4 AO files that include commissioner statements out of the last 30 AOs issued by the commission. Will that be confusing to new users?

For the date range—does that apply only to the document the user filters for? Is there an “all” category? If so, what document does the date range apply to? In our current search, folks can search a date range by request date. Is that still an easy option?

I think “participant” doesn’t capture the roles as well. It is also a term of art in some of our other procedures, so I’d suggest sticking with entities.

Thanks!

**Comment by emileighoutlaw** [comment]
Thanks for the amazingly thoughtful feedback, Amy!

**Document type**: I'm not sure about the decision-making that went into ordering document types and weighing which should have checkboxes vs. which appear in the drop-down (I definitely _should_ know, since this is a content question, but I'm going to have to ask maybe @anthonygarvan or someone else to weigh in?)

**Date range**: Again, I'm not sure :/

**Participant**: We can definitely change this back to `entity`. As a follow-up question for @amykort, if `entity` and `participant` are terms of art, should add them to our glossary? Also— we're have `participant` as a filter for MURs (and not `entity`). Would `entity` be more accurate there, too?

**Comment by jenniferthibault** [comment]
I think I can help on the remaining two!

**Document type:** @nickykrause reached out to Cheryl (who in turn surveyed some FEC policy folks) to see which AO filters were the most used to narrow down their results in the current system. You can read along in the results in [issue #274 's comment](https://github.com/18F/fec-eregs/issues/274#issuecomment-252114573).

We need to surface `Final opinion` as an applied filter since you're first searching only final opinions (based on recent changes requested to do so).

We then chose `Commissioner statements` and `Draft documents` to surface out of the drop-down because they were the next most common document types used by those people. This is similar to how we surface `Republican Party` and `Democratic Party` in the [Candidate data filters](https://beta.fec.gov/data/candidates/)
<img width="296" alt="screen shot 2016-11-01 at 5 47 55 pm" src="https://cloud.githubusercontent.com/assets/11636908/19908858/6e82dc12-a05b-11e6-8b82-bc311e5ee787.png">

**Choice from here:**

1. Move `Commissioner statements` and `Draft documents` back into the dropdown, only keeping `Final opinions` out since they're pre-filtered by default. Test to see if people commonly use other specific options.
2. Replace `Commissioner statements` and `Draft documents` with other, more helpful options outside the dropdown if we have evidence of which they are now. 
3. Keep as is, and test to see if those categories are used frequently enough to keep them surfaced.

**Date range:** I think I may know what happened here. It looks like in [Nicky's same comment](https://github.com/18F/fec-eregs/issues/274#issuecomment-252114573) from her discussion with Cheryl, the AO Request Date was bulleted as a sub-point under document type instead of it's own field, and we didn't catch it:

<img width="265" alt="screen shot 2016-11-01 at 5 54 00 pm" src="https://cloud.githubusercontent.com/assets/11636908/19909031/359ae7cc-a05c-11e6-812d-f7f0fddb4e77.png">

It _is_ still useful to notice that 2/5 FEC folks reported using the `AO Issue Date` field, but 0/5 reported using the `AO Request Date`field. This later leads to @nickykrause 's next comment where we recommend `Issue Date` as one of the first set of filters to focus on. 

**Choice from here:**

1. Implement `Issue date` only for now, and see if people ask for request date through testing, or if Issue Date is sufficient in helping them get to a sufficiently narrowed number of results.
  - Pros: smaller set of filters to implement in first pass
  - Cons: may remove existing functionality (but with small data questioning use of existing functionality)
2. Implement both `Issue date` and `Request date` in the first round. See if people use both in testing. 
  - Pros: keeps existing functionality
  - Cons: may be supporting functionality that isn't needed

**Comment by AmyKort** [comment]
Thanks to you both. Let me talk this through with folks over here and get back to you tomorrow.

**Comment by jenniferthibault** [comment]
Here's one idea for a tweak that came from learning more about pending AO Requests today:

We have the ability to find pending AO requests, and ideally would like that to be part of the unified search, so that you can search all types of AOs in one place.

So instead of surfacing "Commissioner statements" out of the dropdown, we could surface document types that have high use rates and are similar to "Final opinion". This makes "Pending requests" easier to find, and groups it with other AO lifecycle related options, like Final & Drafts.

![screen shot 2016-11-02 at 4 59 21 pm](https://cloud.githubusercontent.com/assets/11636908/19947360/19bfa174-a11e-11e6-9586-eceb4fcd5c81.png)

**Comment by AmyKort** [comment]
I like it! Let me see what other folks think.

**Comment by AmyKort** [comment]
Cheryl's thoughts: For the filters, can we make it  "Final opinion" like it is now, then "Draft documents" and then "Requests" and then everything else after that goes under "More"?

Then she was asking if we could separate Pending AORs into a separate box. She worries it will be confusing to folks to have the pending materials mixed in with the completed files. This would also allow us to more easily keep the files together (pending AORs have drafts and comments, too). How complicated is that?

**Comment by jenniferthibault** [comment]
@AmyKort not too complicated! But I'd like to try out a hypothesis before moving Pending AORs completely out of the document group.

My hypothesis is that users don't think of Pending AOR documents as in a different category from AOs , but rather they think of them as an AO in an earlier state of its lifecycle—one that is active, or in progress, and won't have a final opinion issued. For this reason, grouping them closely will make them easier for users to keep tabs on. 

So the first thing I'd like to test (with your ok) is this arrangement, where Pending is always surfaced but a little separate from the rest of the documents. We could also use an icon in the results list to make them visually a little different too, much like what we do with archived MUR cases:

<img width="261" alt="screen shot 2016-11-04 at 12 00 41 pm" src="https://cloud.githubusercontent.com/assets/11636908/20016751/6a68a26a-a286-11e6-9aa9-e9f53b0fe4f1.png">

If that seems ok, I'd flag this as ready for implementation, and we'd test that arrangement with users, then make any changes to reduce any questions or confusion that arises through testing.

Looking back at the document types, I have just one question on Chery's doc list:

- should `Requests` be separated from `Requests, Supplemental Material, and Extensions of Time`?

**Comment by AmyKort** [comment]
On document type:  That's all one thing, but the name is really long. Hmmm. 

On the proposal. Let's try it and see what people think. One possible tweak would be to put the pending requests under the other group. That ranks them out of chronological order, but the universe of final opinions is much bigger and I think more likely to be sought than the universe of pending requests.

**Comment by emileighoutlaw** [comment]
Is it possible to shorten  `Requests, Supplemental Material, and Extensions of Time` to `Requests`? 

I think the answer lies in whether users regularly come to the search with the phrase "supplemental material" and "extensions of time" in mind. 

If so, then they'll be scanning for words that aren't there, if we shorten it to `Requests`, and that's a problem.

If not, this becomes a question of whether it's legally incorrect or misleading to call the section `Requests`. I don't think so, but I defer to you smart, FEC folks to know the intricacies of this.

**Comment by AmyKort** [comment]
I was also thinking of trying to shorten it to requests. Or, what about "Request Documents"?

**Comment by emileighoutlaw** [comment]
`Request documents` sounds good to me, and it makes a nice parallel with `Draft documents`

**Comment by jenniferthibault** [comment]
Sounds good to me! Thanks so much for all the thinking around this!

To summarize, here's what our Document Type section should look like for the first round of implementation and testing:

![screen shot 2016-11-07 at 4 02 31 pm](https://cloud.githubusercontent.com/assets/11636908/20075781/ac40dc96-a503-11e6-85e6-ebc8a190b00b.png)


I'm going to move this into the "Ready" column for implementation.

**Comment by AmyKort** [comment]
Do you worry that in this version the "Request Documents" and the "Pending Requests" are confusing? Did we lose the little hourglass? (I loved the little hourglass.) What about the idea to put pending requests under a bar?

**Comment by jenniferthibault** [comment]
Oh! @AmyKort I think I misinterpreted what you meant by `One possible tweak would be to put the pending requests under the other group.` Sorry about that, I'll bring the Pending requests back out. Thanks for speaking up!

I feel good about understanding if there's any confusion in the new filter names through showing this to users. We can easily pivot and revise them based on the feedback from testing. 

The icon's hanging around, promise! I should have mentioned that I'm playing around with having it appear in the results list rather than the filter panel, much like we do with MURs. I think I'm going to track that separately from this filter issue:

<img width="250" alt="screen shot 2016-11-07 at 4 13 32 pm" src="https://cloud.githubusercontent.com/assets/11636908/20076114/2fcdbeac-a505-11e6-98e3-1b525160d0ea.png">

**Comment by jenniferthibault** [comment]
Annnd the screenshot I meant to include:

![screen shot 2016-11-07 at 4 16 22 pm](https://cloud.githubusercontent.com/assets/11636908/20076207/8cbd6950-a505-11e6-8339-c914ff2c0efa.png)

**Comment by AmyKort** [comment]
Awesome! That was my fault for not being clear. Thank you @jenniferthibault ! Looks great!

**Comment by noahmanger** [comment]
I just broke this out into separate issue:
- https://github.com/18F/openFEC/issues/2056
- https://github.com/18F/openFEC-web-app/issues/1705
- https://github.com/18F/openFEC/issues/2057

So I'm closing this.
